app/Http/Controllers/DiagnosticsController.php

```diff
@@ -134,8 +134,22 @@ public function get_errors()
 				$errors[] = 'Error: PHP ' . $extension . ' extension not activated';
 			}
 		}
-		if (!extension_loaded('mysqli') && !DB::getDriverName() == 'pgsql') {
-			$errors[] = 'Error: PHP mysqli extension not activated';
+
+		$db_possibilities = [
+			['mysql', 'mysqli'],
+			['mysql', 'pdo_mysql'],
+			['pgsql', 'pgsql'],
+			['pgsql', 'pdo_pgsql'],
+			['sqlite', 'sqlite3'],
+		];
+		try {
+			foreach ($db_possibilities as $db_possibility) {
+				if (DB::getDriverName() == $db_possibility[0] && !extension_loaded($db_possibility[1])) {
+					$errors[] = 'Error: PHP ' . $db_possibility[1] . ' extension not activated';
```

Comment:
For clarity, maybe this? Might need line wrapping.
```suggestion
					$errors[] = 'Error: ' . $db_possibility[0] . ' db driver selected and PHP ' . $db_possibility[1] . ' extension not activated';
```

Reply:
I don't think I'd looked at these properly before. Maybe I should use them more often for the easy-commit, especially as we squash-merge.